runners/reference/java/src/main/java/org/apache/beam/runners/reference/testing/TestJobService.java

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.reference.testing;
+
+import io.grpc.stub.StreamObserver;
+import org.apache.beam.model.jobmanagement.v1.JobApi.CancelJobRequest;
+import org.apache.beam.model.jobmanagement.v1.JobApi.CancelJobResponse;
+import org.apache.beam.model.jobmanagement.v1.JobApi.GetJobStateRequest;
+import org.apache.beam.model.jobmanagement.v1.JobApi.GetJobStateResponse;
+import org.apache.beam.model.jobmanagement.v1.JobApi.JobMessagesRequest;
+import org.apache.beam.model.jobmanagement.v1.JobApi.JobMessagesResponse;
+import org.apache.beam.model.jobmanagement.v1.JobApi.JobState;
+import org.apache.beam.model.jobmanagement.v1.JobApi.PrepareJobRequest;
+import org.apache.beam.model.jobmanagement.v1.JobApi.PrepareJobResponse;
+import org.apache.beam.model.jobmanagement.v1.JobApi.RunJobRequest;
+import org.apache.beam.model.jobmanagement.v1.JobApi.RunJobResponse;
+import org.apache.beam.model.jobmanagement.v1.JobServiceGrpc.JobServiceImplBase;
+import org.apache.beam.model.pipeline.v1.Endpoints.ApiServiceDescriptor;
+
+/**
+ * A JobService for tests.
+ *
+ * <p>A {@link TestJobService} always returns a fixed staging endpoint, job preparation id, job id,
+ * and job state. As soon as a job is run, it is put into the given job state.
+ */
+public class TestJobService extends JobServiceImplBase {
+
+  private final ApiServiceDescriptor stagingEndpoint;
+  private final String preparationId;
+  private final String jobId;
+  private final JobState.Enum jobState;
+
+  public TestJobService(
+      ApiServiceDescriptor stagingEndpoint,
+      String preparationId,
+      String jobId,
+      JobState.Enum jobState) {
+    this.stagingEndpoint = stagingEndpoint;
+    this.preparationId = preparationId;
+    this.jobId = jobId;
+    this.jobState = jobState;
+  }
+
+  @Override
+  public void prepare(
+      PrepareJobRequest request, StreamObserver<PrepareJobResponse> responseObserver) {
+    responseObserver.onNext(
+        PrepareJobResponse.newBuilder()
+            .setPreparationId(preparationId)
+            .setArtifactStagingEndpoint(stagingEndpoint)
+            .build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void run(RunJobRequest request, StreamObserver<RunJobResponse> responseObserver) {
+    responseObserver.onNext(RunJobResponse.newBuilder().setJobId(jobId).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void getState(
+      GetJobStateRequest request, StreamObserver<GetJobStateResponse> responseObserver) {
+    responseObserver.onNext(GetJobStateResponse.newBuilder().setState(jobState).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void cancel(CancelJobRequest request, StreamObserver<CancelJobResponse> responseObserver) {
+    super.cancel(request, responseObserver);
```

Comment:
Remove overrides that are exactly equal to the `super` method

Reply:
Done.